src/vaadin-checkbox.html

```diff
@@ -196,7 +196,12 @@
 
         _updateLabelAttribute() {
           const label = this.shadowRoot.querySelector('[part~="label"]');
-          if (label.firstElementChild.assignedNodes().length === 0) {
+          const assignedNodes = label.firstElementChild.assignedNodes();
+          // The label is empty if there is no slotted content or only one empty text node
+          if (assignedNodes.length === 0 ||
+              (assignedNodes.length == 1
+                && assignedNodes[0].nodeType == 3
+                && assignedNodes[0].textContent.trim().length === 0)) {
```

Comment:
`textContent.trim().length === 0` can be replaced with `textContent.trim() === ''`, it's a bit shorted and a bit more readable. not a blocker though.